src/Json.php

```diff
@@ -103,6 +103,8 @@ public static function encode($valueToEncode, $cycleCheck = false, $options = []
                 return $valueToEncode->toJson();
             } elseif (method_exists($valueToEncode, 'toArray')) {
                 return static::encode($valueToEncode->toArray(), $cycleCheck, $options);
+            } elseif ($valueToEncode instanceof ArrayIterator) {
```

Comment:
Considering the implications of PHP converting the iterator's keys with the second parameter?

-----Original Message-----
From: "Maks3w" <notifications@github.com>
Sent: ‎11/‎25/‎2015 12:43 PM
To: "zendframework/zend-json" <zend-json@noreply.github.com>
Cc: "robob4him" <robob4him@gmail.com>
Subject: Re: [zend-json] Add support for ArrayIterator (#6)

In src/Json.php:
> @@ -103,6 +103,8 @@ public static function encode($valueToEncode, $cycleCheck = false, $options = []
>                  return $valueToEncode->toJson();
>              } elseif (method_exists($valueToEncode, 'toArray')) {
>                  return static::encode($valueToEncode->toArray(), $cycleCheck, $options);
> +            } elseif ($valueToEncode instanceof ArrayIterator) {
ArrayInput have a transitive implementation of Iterator interface which implements Traversable. So this could be more generic and use iterator_to_array for transform any Traversable object
—
Reply to this email directly or view it on GitHub.